server/impl/src/main/java/com/walmartlabs/concord/server/events/GithubEventResource.java

```diff
@@ -193,18 +194,30 @@ public Map<String, Object> resolve(TriggerEntry t) {
             }
 
             GithubTriggerExclusiveMode e = objectMapper.convertValue(exclusive, GithubTriggerExclusiveMode.class);
-            if (e.groupBy() == null) {
+            String groupBy = e.groupBy();
+            if (groupBy== null) {
                 return exclusive;
             }
 
             String group;
-            switch (Objects.requireNonNull(e.groupBy())) {
-                case branch: {
-                    group = payload.getBranch();
-                    break;
+            if ("branch".equals(groupBy)) {
+                group = payload.getBranch();
+            } else if (groupBy.startsWith("event")) {
+                String[] payloadPath = groupBy.split("\\.");
+                if (payloadPath.length == 1) {
+                    throw new IllegalArgumentException("Invalid groupBy: '" + e.groupBy() + "'");
                 }
-                default:
-                    throw new IllegalArgumentException("Unknown groupBy: '" + e.groupBy() + "'");
+
+                payloadPath = Arrays.copyOfRange(payloadPath, 1, payloadPath.length);
+                Object maybeString = ConfigurationUtils.get(payload.raw(), payloadPath);
+                if (maybeString == null || (maybeString instanceof String)) {
+                    group = (String) maybeString;
```

Comment:
I'm trying to wrap my head around what happens when the value is `null`. Does that effectively not group it? Or is `null` a valid group and possibly a bunch of processes get grouped (and `cancel`led or `wait`ed)?

Reply:
logic will be the same as before, if group is null then no exclusive attributes for process:

https://github.com/walmartlabs/concord/blob/brig/gh-trigger-groupby-event-attr/server/impl/src/main/java/com/walmartlabs/concord/server/events/GithubEventResource.java#L223